### ISSUE TITLE
system.properties in filesystem root

### ISSUE THREAD

**Issue by Programie** [jmkeyes/puppet-unifi#2]
Using the unifi::setting define results in a system.properties in the filesystem root as $data_directory does not exist in the unifi class (resulting in the file path "/system.properties").

**Comment by jmkeyes** [comment]
Good catch! I've pushed commit 866abca on the master branch that should fix it.

Can you try out that change and let me know how it goes?

**Comment by Programie** [comment]
I haven't tested it yet, but as far as I know, you have to move $data_dir in the unifi class to the class parameters.

**Comment by jmkeyes** [comment]
Good point. I've made them into parameters of `Class[unifi]`. I've also noticed that the Unifi Controller package now stores most of it's data, logs, and runtime information under subdirectories of `/var`, so I've updated the module to take the new paths into account. (f1f872f7)

I've also added a `require` parameter to the `Unifi::Setting` defined type so that Puppet applies them after `Package[unifi]` is installed. (d51b97fb)
